pkg/apis/camel/v1/maven_types.go

```diff
@@ -30,16 +32,17 @@ type MavenSpec struct {
 	// A reference to the ConfigMap or Secret key that contains
 	// the Maven settings.
 	Settings ValueSource `json:"settings,omitempty"`
-	// The Secret name and key, containing the CA certificate(s) used to connect
+	// The Secrets name and key, containing the CA certificate(s) used to connect
 	// to remote Maven repositories.
 	// It can contain X.509 certificates, and PKCS#7 formatted certificate chains.
 	// A JKS formatted keystore is automatically created to store the CA certificate(s),
 	// and configured to be used as a trusted certificate(s) by the Maven commands.
 	// Note that the root CA certificates are also imported into the created keystore.
-	CASecret *corev1.SecretKeySelector `json:"caSecret,omitempty"`
+	CASecret []corev1.SecretKeySelector `json:"caSecret,omitempty"`
 	// The Maven build extensions.
 	// See https://maven.apache.org/guides/mini/guide-using-extensions.html.
 	Extension []MavenArtifact `json:"extension,omitempty"`
+	Servers   []Server        `json:"servers,omitempty"`
```

Comment:
Can that be also provided in the user Maven settings?

Reply:
Hmm not sure what you mean by user Maven settings ? During installation ?

`kamel install --maven-settings=..`

If so yes

Reply:
I mean configured in the Maven settings that the users can already provide?

Reply:
Right so referenced in a `secret/cm` ? Typically during installation:

`kamel install --maven-settings=mvnsettingsconfigmap`

Yes the server tags are a standard part of Maven user settings. I have added it in the maven_types so we can generate these new tags based on the registry secret. But I'm probably misunderstanding your question :D

Reply:
I'm asking because we've maintained strict partitioning between what goes into the settings and what goes into the API to avoid duplication. So if the servers configuration can be done in the settings, it should be removed from the API. So we can instruct users to do so in their custom Maven settings file, and/or configure the servers in the platform settings file that is generated for each build.

Reply:
I don't think the CA certificates can be provided. The APIs pre-dates the realisation that the Maven configuration should be centralised in the Maven settings, hence the present fields, but I think we should aim towards having only one place to configure things.

Reply:
Oh right I get what you are saying. 

I didn't really mean to be part of the 'API' (the Maven Servers aren't configurable via the Kamel CLI for example).

I just wanted the camel-k operator to retrieve the already configured Image Pull Secret and convert it into a [Maven Server](https://github.com/apache/camel-k/blob/36ec1d005ebb91d2783a8fc6c47cfe117f369b20/pkg/trait/registry.go#L104). That way the user doesn't have to configure it twice.

Reply:
Yes, if you need to configure the build, the `Servers` field is better to be moved in the Build API, instead of being exposed in the IntegrationPlatform API:

https://github.com/apache/camel-k/blob/7942108bf0e00b322580387773eb7222bc6c0d18/pkg/apis/camel/v1/build_types.go#L85-L90

Reply:
Ok will do thanks!